### PR TITLE
Support custom modifier separator

### LINKED ISSUE #69: Provide a way to support custom modifier separator
like `block__elem_mod!!!val`

For now it's impossible to do.

Purpose is using for Harry Roberts' notation: `block__elem--mod_val`.

### PR DESCRIPTION
**Pull request by blond**
Resolved #69 
- Add `modValDelim` field.
- Support `mod` as object: `mod: { name: String, val: String }`.
- Update convention by Harry Roberts: use `mod: { name: '--', val: '_' }`.

### CODE REVIEW

**Comment by tadatuta** [comment]
lgtm

**Comment by blond** [comment]
:up:

**Comment by tadatuta** [comment]
still lgtm :)

File: README.md [hunk 1]
```diff
@@ -325,22 +332,33 @@ Use `bemNaming` function to create instance to manage naming of your own naming
...
+  This option can get object:
+
+  ```js
+  { name: String, val: String }
+  ```
+
+  * **String** `name` — separates name of modifier from blocks and elements. Default as `_`.
+  * **String** `val` — separates value of modifier from name of modifier. Default as the value of the `name`.
+
+* **String** `modVal` — separates value of modifier from name of modifier. Default as the value of the `mod` option.
```
Comment (zxqfox) [inline README.md]: It's looks like neprisheikpeacedeadrookave. Do we really need it?

File: README.md [hunk 2]
```diff
@@ -325,22 +332,33 @@ Use `bemNaming` function to create instance to manage naming of your own naming
 
 Function `bemNaming` gets the object from the following options:
 
-* **String** `elem` — separates element's name from block. Default&nbsp;as&nbsp;`__`.
-* **String** `mod` — separates names and values of modifiers from blocks and elements. Default&nbsp;as&nbsp;`_`.
-* **String** `wordPattern` — defines which symbols can be used for block, element and modifier's names. Default&nbsp;as&nbsp;`[a-z0-9]+(?:-[a-z0-9]+)*`.
+* **String** `elem` — separates element's name from block. Default as `__`.
+* **String** `mod` — separates modifiers from blocks and elements. Default as `_`.
```
Comment (zxqfox) [inline README.md]: String|{name:String, val:String}

**Comment by blond** [comment]
@zxqfox I removed `modVal` option.

**Comment by zxqfox** [comment]
:+1: ok, thanks ;-)

**Comment by seyfer** [comment]
@blond 
Hey, somebody could explain me, please, why more readable modifier delimiter `--` replaced with `_`?
Who is Harry Roberts I know, I saw his articles, but I can't find one with the explanation and the decision there https://csswizardry.com/archive/

Also, you can notice that a lot of people not aware of changes and still use old way with `--` in their BEM code. So, what the point?

For example, there it's still `--` http://getbem.com/naming/
But in the docs, it is `_` https://en.bem.info/methodology/naming-convention/

We are implementing BEM now in our team and we are confused.

**Comment by tadatuta** [comment]
@seyfer actually underscore was historically first demimiter originated from Yandex.
Double-dash version was introduced later and was mostly just a matter of taste.

But right now we tend to switch to so called [React style](https://en.bem.info/methodology/naming-convention/#react-style).

Still the main point here is that any particular scheme does not matter while all the code base of a project is consistent.

**Comment by qfox** [comment]
@tadatuta Feels like we should make a faq for naming related questions.

**Pull request merged**